### ISSUE TITLE
Performance issue in caching for remote huggingface models

### ISSUE THREAD

**Issue by sscdotopen** [HazyResearch/manifest#103]
## Description of the bug

Currently, the `HuggingFaceClient` makes an HTTP request in [get_model_params](
https://github.com/HazyResearch/manifest/blob/6324e0fe439723cfc5a9238551562d9f4287e39c/manifest/clients/huggingface.py#L76) for each generation. This seriously slows down execution when the result is already cached (and the underlying HTTP connection is slow), especially in cases where one has to make a larger number of calls.

## Expected behavior

If I understand correctly, the result of `get_model_params` is static, e.g.:

```
{'model_name': 'togethercomputer/RedPajama-INCITE-Instruct-3B-v1',
 'model_path': 'togethercomputer/RedPajama-INCITE-Instruct-3B-v1',
 'client_name': 'huggingface'}
```

Therefore, my proposal would be to simply cache these params after one initial request and thereby get rid of the overhead for later requests. I hacked this into my code and it seems work well:

```python
from manifest import Manifest 
# Serving https://huggingface.co/togethercomputer/RedPajama-INCITE-Instruct-3B-v1 via SSH portforwarding 
redpajama_client = Manifest(client_name = "huggingface", client_connection = "http://127.0.0.1:5550",
                            cache_name='sqlite', cache_connection="demo_data/rp3b-cache.sqlite")

# Hack to cache model params
from manifest.clients.huggingface import HuggingFaceClient
import types

client = redpajama_client.client_pool.get_current_client()
redpajama_model_params = client.get_model_params()

def cached_params(self):
    return redpajama_model_params

client.get_model_params = types.MethodType(cached_params, client)

```

**Comment by lorr1** [comment]
I can also add a little cache decorator. But good suggestion. I'll try to push a fix in the next few days.

**Comment by sscdotopen** [comment]
Thank you!

**Comment by lorr1** [comment]
Merged to main with #105
